### PR TITLE
feat(vercel): ESM support and Node.js 16

### LINKED ISSUE #4782: Update Vercel adapter to Node 14.8 to fix top level await builds
### Describe the bug

SvelteKit has been updated to `node14.8` to enable top-level await per #4742 .  Would be great if we could update the [vercel adapter](https://github.com/sveltejs/kit/blob/master/packages/adapter-vercel/index.js) to the same version since it is currently forcing build version `node14` and there doesn't appear to be a way to override it

### Reproduction

repo [here](https://github.com/digitaldeus/vercel-top-level-await)

to repro just run:
```
npm install
npm run build
```

### Logs

```shell
> Using @sveltejs/adapter-vercel
✘ [ERROR] Top-level await is not available in the configured target environment ("node14")

    .svelte-kit/output/server/chunks/hooks-45a47dbd.js:1:0:
      1 │ await Promise.resolve(true);
        ╵ ~~~~~

> Build failed with 1 error:
.svelte-kit/output/server/chunks/hooks-45a47dbd.js:1:0: ERROR: Top-level await is not available in the configured target environment ("node14")
.svelte-kit/output/server/chunks/hooks-45a47dbd.js:1:0: ERROR: Top-level await is not available in the configured target environment ("node14")
    at failureErrorWithLog (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1603:15)
    at /Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1249:28
    at runOnEndCallbacks (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1162:65)
    at buildResponseToResult (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1247:7)
    at /Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:1356:14
    at /Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:666:9
    at handleIncomingPacket (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:763:9)
    at Socket.readFromStdout (/Users/deusduke/src/vercel-top-level-await/node_modules/esbuild/lib/main.js:632:7)
    at Socket.emit (node:events:527:28)
    at addChunk (node:internal/streams/readable:324:12)
```

### System Info

```shell
System:
    OS: macOS 12.3.1
    CPU: (8) arm64 Apple M1
    Memory: 96.47 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 17.8.0 - /opt/homebrew/bin/node
    Yarn: 1.22.18 - ~/.yarn/bin/yarn
    npm: 8.5.5 - /opt/homebrew/bin/npm
  Browsers:
    Brave Browser: 100.1.37.116
    Chrome: 100.0.4896.127
    Safari: 15.4
```

### Severity

blocking all usage of SvelteKit

### Additional Information

Unable to use sveltekit in vercel due to top level await issue

### PR DESCRIPTION
**Pull request by JuanM04**
Fixes #4782

This PR adds support for ESM deployments to Vercel by adding a package.json with `{ "type": "module" }` ([AWS post about this](https://aws.amazon.com/blogs/compute/using-node-js-es-modules-and-top-level-await-in-aws-lambda/)). ~~It also allows the user to select which Node.js version they want to use (it can be `16.x` (default), `14.x` or `12.x`).~~ It also allows deploying to Node.js 16, which can be configure inside the Vercel Project's settings

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/adapter-vercel/index.d.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ type Options = {
 	edge?: boolean;
 	external?: string[];
 	split?: boolean;
+	nodeVersion?: '16.x' | '14.x' | '12.x';
```
Comment (benmccann) [inline packages/adapter-vercel/index.d.ts:7]: why support Node 12? we don't anywhere else

how about Node 18?

Comment (JuanM04) [reply packages/adapter-vercel/index.d.ts:7]: Node 18 isn't support by Vercel :disappointed: 

If Svelte doesn't support node 12.22, I can remove it

**Comment by Rich-Harris** [comment]
Thanks — can we auto-detect the node version? The same version is used during build as in the deployment, so we should be able to get rid of the option altogether and Do The Right Thing based on `process.version` (including throwing an error if it's 12)

**Comment by JuanM04** [comment]
@Rich-Harris Done!

**Review by Rich-Harris — APPROVED** [review]

**Comment by Rich-Harris** [comment]
thank you!

**Comment by jacob-8** [comment]
I will post an issue when I have a useful repro, but I did want to mention now in case anyone else has this issue, that this PR and the resulting bump of `@sveltejs/adapter-vercel@1.0.0-next.50` (`@sveltejs/adapter-auto@1.0.0-next.41`) to `@sveltejs/adapter-vercel@1.0.0-next.51` (`@sveltejs/adapter-auto@1.0.0-next.42`) broke my use of Firebase in all my apps whenever I update. On Vercel deployments (but not local dev nor preview), I now receive this error: `500 Failed to fetch dynamically imported module: https://<app-url-here>.vercel.app/_app/undefined`. The deployment succeeds but the app fails when it gets to initing Firebase. I can avoid the problem by staying at `@sveltejs/adapter-auto@1.0.0-next.41` and `@sveltejs/kit@1.0.0-next.330`.

**Pull request merged**